tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/auth/OAuthConfigTest.java

```diff
@@ -0,0 +1,85 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.ballerinalang.test.auth;
+
+import org.ballerinalang.test.context.BMainInstance;
+import org.ballerinalang.test.context.LogLeecher;
+import org.testng.annotations.Test;
+
+import java.nio.file.Paths;
+
+/**
+ * Testing OAuth Client Authentication Configs.
+ */
+@Test(groups = "auth-test")
+public class OAuthConfigTest extends AuthBaseTest {
+
+    private BMainInstance ballerinaClient;
+    private static final String BAL_FILE = Paths.get("src", "test", "resources", "auth",
+            "authclients", "oauth-client.bal").toFile().getAbsolutePath();
+
+    @Test(description = "Test client authentication sent as body param")
+    public void testClientAuthenticationAsBodyParam() throws Exception {
+        final String serverResponse = "{\"clientIdInBody\":true, \"hasAuthHeader\":false}";
+
+        LogLeecher serverLeecher = new LogLeecher(serverResponse);
+        serverInstance.addLogLeecher(serverLeecher);
+
+        ballerinaClient = new BMainInstance(balServer);
+        ballerinaClient.runMain(BAL_FILE, null, new String[]{"REQUEST_BODY"});
+        serverLeecher.waitForText(20000);
+    }
+
+    @Test(description = "Test client authentication sent as authentication header")
+    public void testClientAuthenticationAsHeader() throws Exception {
+        final String serverResponse = "{\"clientIdInBody\":false, \"hasAuthHeader\":true}";
+
+        LogLeecher serverLeecher = new LogLeecher(serverResponse);
+        serverInstance.addLogLeecher(serverLeecher);
+
+        ballerinaClient = new BMainInstance(balServer);
+        ballerinaClient.runMain(BAL_FILE, null, new String[]{"BASIC_AUTH_HEADER"});
+        serverLeecher.waitForText(20000);
+    }
+
+    @Test(description = "Test client authentication sent as authentication header by default")
+    public void testClientAuthenticationDefaultToHeader() throws Exception {
+        final String serverResponse = "{\"clientIdInBody\":false, \"hasAuthHeader\":true}";
+
+        LogLeecher serverLeecher = new LogLeecher(serverResponse);
+        serverInstance.addLogLeecher(serverLeecher);
+
+        ballerinaClient = new BMainInstance(balServer);
+        ballerinaClient.runMain(BAL_FILE, null, new String[]{"NO_CONFIG"});
+        serverLeecher.waitForText(20000);
+    }
+
+    @Test(description = "Test sending scope param in request body")
+    public void testScopeConfig() throws Exception {
+        final String serverResponse = "{scope: true}";
+
+        LogLeecher serverLeecher = new LogLeecher(serverResponse);
+        serverInstance.addLogLeecher(serverLeecher);
+
+        ballerinaClient = new BMainInstance(balServer);
+        ballerinaClient.runMain(BAL_FILE, null, new String[]{"SCOPE"});
+        serverLeecher.waitForText(20000);
+    }
```

Comment:
Being pessimistic here. Since, we are adding stuff to POST body with the "http:POST_BODY_BEARER", it would be better to have an example where you add "scope" to a request that is using "http:POST_BODY_BEARER" "credentialBearer".